spec/requests/posts_request_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'Controllers', type: :request do
+  describe 'PostsController' do
+    describe '#index' do
+      it 'returns a successful response' do
+        get '/users/1/posts'
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'renders the index template' do
+        get '/users/1/posts'
+        expect(response).to render_template(:index)
+      end
+
+      it 'includes correct placeholder text in the response body' do
+        get '/users/1/posts'
+        expect(response.body).to include('List of all posts and comment of a user')
+      end
+    end
+
+    describe '#show' do
+      it 'returns a successful response' do
+        get '/users/1/posts/4'
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'renders the show template' do
+        get '/users/1/posts/4'
+        expect(response).to render_template(:show)
+      end
+
+      it 'includes correct placeholder text in the response body' do
+        get '/users/1/posts/4'
+        expect(response.body).to include('>Details of a selected post')
+      end
+    end
```

Comment:
- [ ] The 3 tests in this block are failing ❌ unfortunately. Please try to make them green ✔️ 
Kindly see the attached image file below to see the errors encountered while running your specs 👇
![image](https://github.com/SaneMyburg/blog-app/assets/46839548/3334cfbb-acf8-472b-99b0-29bb1d42b977)